library/Haste/EventListener/PageListener.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Haste\EventListener;
+
+use Contao\Environment;
+use Contao\Input;
+use Haste\Util\AjaxReloadHelper;
+
+class PageListener
+{
+    /**
+     * On generate the page
+     */
+    public function onGenerate()
+    {
+        if (!AjaxReloadHelper::hasListeners()) {
+            return;
+        }
+
+        $GLOBALS['TL_JAVASCRIPT'][] = 'system/modules/haste/assets/ajax-reload.min.js';
```

Comment:
You should use the Haste Debug class to include that file (which removes .min in debug ode)